Application/Api/Education/ClassRegister/ApiDiary.php

```diff
@@ -0,0 +1,585 @@
+            self::API_TARGET => 'openEditDiaryModal',
+        ));
+        $ModalEmitter->setPostPayload(array(
+            'DiaryId' => $DiaryId
+        ));
+        $Pipeline->appendEmitter($ModalEmitter);
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param $DiaryId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineEditDiarySave($DiaryId)
+    {
+        $Pipeline = new Pipeline();
+        $ModalEmitter = new ServerEmitter(self::receiverModal(), self::getEndpoint());
+        $ModalEmitter->setGetPayload(array(
+            self::API_TARGET => 'saveEditDiaryModal'
+        ));
+        $ModalEmitter->setPostPayload(array(
+            'DiaryId' => $DiaryId
+        ));
+        $ModalEmitter->setLoadingMessage('Wird bearbeitet');
+        $Pipeline->appendEmitter($ModalEmitter);
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param int $DiaryId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineOpenDeleteDiaryModal($DiaryId)
+    {
+        $Pipeline = new Pipeline(false);
+        $ModalEmitter = new ServerEmitter(self::receiverModal(), self::getEndpoint());
+        $ModalEmitter->setGetPayload(array(
+            self::API_TARGET => 'openDeleteDiaryModal',
+        ));
+        $ModalEmitter->setPostPayload(array(
+            'DiaryId' => $DiaryId
+        ));
+        $Pipeline->appendEmitter($ModalEmitter);
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param $DiaryId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineDeleteDiarySave($DiaryId)
+    {
+
+        $Pipeline = new Pipeline();
+        $ModalEmitter = new ServerEmitter(self::receiverModal(), self::getEndpoint());
+        $ModalEmitter->setGetPayload(array(
+            self::API_TARGET => 'saveDeleteDiaryModal'
+        ));
+        $ModalEmitter->setPostPayload(array(
+            'DiaryId' => $DiaryId
+        ));
+        $ModalEmitter->setLoadingMessage('Wird bearbeitet');
+        $Pipeline->appendEmitter($ModalEmitter);
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param int|null $DivisionId
+     * @param int|null $GroupId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineOpenSelectStudentModal($DivisionId = null, $GroupId = null)
+    {
+        $Pipeline = new Pipeline(false);
+        $ModalEmitter = new ServerEmitter(self::receiverModal(), self::getEndpoint());
+        $ModalEmitter->setGetPayload(array(
+            self::API_TARGET => 'openSelectStudentModal',
+        ));
+        $ModalEmitter->setPostPayload(array(
+            'DivisionId' => $DivisionId,
+            'GroupId' => $GroupId
+        ));
+        $Pipeline->appendEmitter($ModalEmitter);
+
+        return $Pipeline;
+    }
+
+    /**
+     * @param int|null $DivisionId
+     * @param int|null $GroupId
+     *
+     * @return Pipeline
+     */
+    public static function pipelineSelectStudentSave($DivisionId = null, $GroupId = null)
```

Comment:
Codacy found an issue: [Incorrect spacing between argument "$DivisionId" and equals sign; expected 0 but found 1](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)